### PR TITLE
Fix Inventory/Project rbac broken on JT form

### PR DESCRIPTION
**Pull request by nixocio**
Fix Inventory/Project rbac broken on JT form.

Also, update ProjectLookup to filter using `role_level: 'use_role'` as
per old UI implementation.

See: https://github.com/ansible/awx/issues/8194

### CODE REVIEW

File: awx/ui_next/src/components/Lookup/InventoryLookup.jsx [hunk 1]
```diff
@@ -29,6 +29,7 @@ function InventoryLookup({
   fieldId,
   promptId,
   promptName,
+  overrideIsDisabled,
```
Comment (jakemcdermott) [inline awx/ui_next/src/components/Lookup/InventoryLookup.jsx]: nitpick: The convention we usually follow would make this `isOverrideDisabled`

**Review by jakemcdermott — APPROVED** [review]

File: awx/ui_next/src/components/Lookup/InventoryLookup.jsx (hunk 1 above)
Comment (nixocio) [reply awx/ui_next/src/components/Lookup/InventoryLookup.jsx]: @jakemcdermott, updated per comment.

**Comment by unlikelyzero** [comment]
Definitely need to add coverage here. Realistically won't get it to until next week

**Comment by nixocio** [comment]
@unlikelyzero, I rebased this PR, and currently e2e tests.

**Review by tiagodread — APPROVED** [review]
Verified that the normal user with permissions can successfully create a JT

**Comment by nixocio** [comment]
After I followed up with @mabashian, I decided to keep this PR open to verify about EDIT of those forms.

**Comment by unlikelyzero** [comment]
@tiagodread do we have an open PR for this?

**Comment by tiagodread** [comment]
> @tiagodread do we have an open PR for this?

https://github.com/ansible/tower-qa/pull/5786

**Review by tiagodread — APPROVED** [review]
Tests covered by https://github.com/ansible/tower-qa/pull/5786

**Comment by nixocio** [comment]
regate

**Pull request merged**